File: 2216-delete-the-middle-node-of-a-linked-list/delete-the-middle-node-of-a-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteMiddle(ListNode head) {
        ListNode dummy = head;
        ListNode p = dummy;
        int count = 0;
        while(dummy!=null){
            count++;
            dummy = dummy.next;
        }
        if(count==1||count==0)
        return null;
        int mid = count/2;
        int index = 0;
        ListNode ans = p;
        while(true){
            if(mid==index+1){
                p.next = p.next.next;
                break;
            }
            p = p.next;
            index++;
        }
        return ans;
        
    }
}

File: 2216-delete-the-middle-node-of-a-linked-list/delete-the-middle-node-of-a-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteMiddle(self, head: Optional[ListNode]) -> Optional[ListNode]:
        dummy = head
        p = dummy
        count = 0
        
        while dummy:
            count += 1
            dummy = dummy.next
        
        if count == 1 or count == 0:
            return None
        
        mid = count // 2
        index = 0
        ans = p
        
        while True:
            if mid == index + 1:
                p.next = p.next.next
                break
            p = p.next
            index += 1
        
        return ans
            